From 2c8ed76665d5536df7d76bdb10a7f2dee92b8082 Mon Sep 17 00:00:00 2001
From: Thomas Sibley <trs@bestpractical.com>
Date: Mon, 22 Jul 2013 22:29:13 -0700
Subject: [PATCH] Avoid linking trailing punctuation in MakeClicky

---
 share/html/Elements/MakeClicky | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/share/html/Elements/MakeClicky b/share/html/Elements/MakeClicky
index dabe3bb..854c93c 100644
--- a/share/html/Elements/MakeClicky
+++ b/share/html/Elements/MakeClicky
@@ -76,12 +76,12 @@ my %actions = (
 my @types = (
     {
         name   => "httpurl",
-        regex  => qr/$RE{URI}{HTTP}{-keep}{-scheme => 'https?'}(?:#\S+)?/,
+        regex  => qr/$RE{URI}{HTTP}{-keep}{-scheme => 'https?'}(?:#\S+)?(?<![.?!,;:])/,
         action => "url",
     },
     {
         name   => "httpurl_overwrite",
-        regex  => qr/$RE{URI}{HTTP}{-keep}{-scheme => 'https?'}(?:#\S+)?/,
+        regex  => qr/$RE{URI}{HTTP}{-keep}{-scheme => 'https?'}(?:#\S+)?(?<![.?!,;:])/,
         action => "url_overwrite",
     },
 );
-- 
1.8.3.2

